server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -809,13 +941,23 @@ public void removeResourcesByShuffleIds(String appId, List<Integer> shuffleIds)
           taskInfo.getCommitLocks().remove(shuffleId);
         }
       }
-      Optional.ofNullable(partitionsToBlockIds.get(appId))
-          .ifPresent(
-              x -> {
-                for (Integer shuffleId : shuffleIds) {
-                  x.remove(shuffleId);
-                }
-              });
+      if (newBitmapStrategyEnabled) {
```

Comment:
It is hard to maintain code with these kind of if...else...else if...

Instead, I strongly recommend to use a policy driven way